pkg/controller/kogitobuild/kogitobuild_controller_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kogitobuild
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitobuild/build"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure/services"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/test"
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sort"
+	"testing"
+	"time"
+)
+
+//TODO: write all sorts of test cases:
+//single sample build
+//build with custom image
+//build with a pre defined image stream (probably handled by KogitoRuntime)
+//multiple builds (with same target KogitoRuntime)
+
+func assertReconcile(t *testing.T, r ReconcileKogitoBuild, instance *v1alpha1.KogitoBuild) (result reconcile.Result) {
+	result, err := r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}})
+	assert.NoError(t, err)
+	assert.NotNil(t, result)
+	return
+}
+
+func assertReconcileMustRequeue(t *testing.T, r ReconcileKogitoBuild, instance *v1alpha1.KogitoBuild) (result reconcile.Result) {
+	result = assertReconcile(t, r, instance)
+	assert.True(t, result.Requeue)
+	return
+}
+
+func assertReconcileMustNotRequeue(t *testing.T, r ReconcileKogitoBuild, instance *v1alpha1.KogitoBuild) (result reconcile.Result) {
```

Comment:
maybe move it to a helper class/pkg so it can be reused in more test files?

Reply:
This is specific to this test suit. If others would also need, we can then think on generalize. I don't like to create generic functions upfront, is not a good practice.